lib/cdo/pardot.rb

```diff
@@ -66,6 +70,11 @@ class Pardot
   class InvalidApiKeyException < RuntimeError
   end
 
+  # Exception to throw to ourselves if Pardot failed with a known error code
+  # and we want to continue the run.
+  class KnownPardotErrorException < RuntimeError
```

Comment:
These types of exceptions aren’t specific to Pardot.  They could apply to any external webservice.  Maybe rename this to `RetryableHTTPException` or `TransientHTTPException`?

Reply:
Interesting point.  I was mainly treating these as "Pardot errors" because of the naming of the existing variable [here](https://github.com/code-dot-org/code-dot-org/pull/28198/files#diff-f5c9198dcd343771dcbf5a7ac3cd2f7aR28).

Reply:
Yup.  I understand you want to minimize code changes since we don’t have tests for Contact Rollups, but I’d also suggest renaming those variables.  This is a generic problem about how an HTTP client should interact with a web service.  HTTP Status codes >=200 and <300 should be considered to be Success.  HTTP Status codes >=500 should be considered transient failures within the 3rd party service and should be retries.  This would apply for any webservice, not just Pardot.